purchase_ext_sst/models/purchase_order.py

```diff
@@ -84,113 +95,96 @@ def open_record(self):
+            val_phone_search = 'phone_search' in vals and vals[
+                'phone_search'] or False
+            val_phone = 'phone_update' in vals and vals['phone_update'] or \
+                        val_phone_search or False
+            val_mobile = 'mobile_update' in vals and vals['mobile_update'] \
                          or False
-        if self.is_default_partner(vals['partner_id']):
-            if val_phone or val_mobile:
-                partner_e = self.get_partner_from_phone(val_phone, val_mobile)
-                if partner_e and vals['partner_id'] != partner_e.id:
-                    vals['partner_id'] = partner_e.id
-                    if not vals['phone']:
-                        vals['phone'] = partner_e.phone
-                    if not vals['mobile']:
-                        vals['mobile'] = partner_e.mobile
-                if not partner_e:
+            val_tent_name = 'tentative_name' in vals and vals[
+                'tentative_name'] or order.tentative_name
+            vals['partner_id'] = 'partner_id' in vals and vals['partner_id']\
+                                 or order.partner_id.id
+            partner = self.get_partner_from_phone(val_phone, val_mobile)
+            if self.is_default_partner(vals['partner_id']):
+                if partner:
+                    vals['partner_id'] = partner.id
+                elif val_phone or val_mobile:
                     vals['partner_id'] = self.create_partner(val_phone,
                                                              val_mobile).id
+            if not self.is_default_partner(vals['partner_id']):
                 partner = self.env['res.partner'].browse(vals['partner_id'])
                 self.update_partner(partner, val_phone, val_mobile,
                                     val_tent_name)
-        else:
-            if val_phone or val_mobile:
-                partner_e = self.get_partner_from_phone(val_phone, val_mobile)
-                if partner_e and vals['partner_id'] != partner_e.id:
-                    vals['partner_id'] = partner_e.id
-                partner = self.env['res.partner'].browse(vals['partner_id'])
-                self.update_partner(partner, val_phone, val_mobile,
-                                    val_tent_name)
+        vals['phone_update'] = vals['mobile_update'] = vals['phone_search'] \
+            = False
+        return super(PurchaseOrder, self).write(vals)
+
+    @api.model
+    def create(self, vals):
+        val_phone_search = 'phone_search' in vals and vals['phone_search'] \
+                           or False
+        val_phone = 'phone_update' in vals and vals['phone_update'] or \
+                    val_phone_search or False
+        val_mobile = 'mobile_update' in vals and vals['mobile_update'] or False
+        val_tent_name = 'tentative_name' in vals and vals['tentative_name'] \
+                         or False
+        partner = self.get_partner_from_phone(val_phone, val_mobile)
```

Comment:
We should not need this if partner is proposed from `onchange_phone_search()` method.